src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1253,15 +1271,15 @@ namespace winrt::TerminalApp::implementation
             auto valueSet = TerminalConnection::ConptyConnection::CreateSettings(settings.Commandline(),
                                                                                  newWorkingDirectory,
                                                                                  settings.StartingTitle(),
+                                                                                 settings.ReloadEnvironmentVariables(),
+                                                                                 _WindowProperties.VirtualEnvVars(),
```

Comment:
Can't we use the existence of a non-empty `VirtualEnvVars()` that `ReloadEnvironmentVariables()` is supposed to be `false`?

Reply:
Alas, no. `VirtualEnvVars` is always going to be filled with something. Either our own env vars, or the ones from the original `wt` that then told us to do something.

Reply:
After reading through it I now get it... There's 3 states:
* Fresh env from registry
* Stale env from PEB
* Virtual env from settings

The boolean is there to distinguish between the 1st and the latter 2 of those.